hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1440 @@
+        if (file == nullptr) {
+            std::cout << "FINEST: "
+                      << "Could not find " + config_file_path +
+                           " in the working directory."
+                      << std::endl;
+            return false;
+        }
+        std::cout << "INFO: Loading " + config_file_path +
+                       " from the working directory."
+                  << std::endl;
+        configuration_file = file;
+        in = new std::ifstream();
+        in->open(config_file_path, std::ios::in);
+        if (in->fail()) {
+            throw hazelcast::client::exception::hazelcast_(
+              "Failed to open file: " + config_file_path);
+        }
+        return true;
+    } catch (const std::runtime_error& e) {
+        throw hazelcast::client::exception::hazelcast_(e.what());
+    }
+}
+
+bool
+abstract_config_locator::load_from_working_directory(
+  const std::string& config_file_prefix,
+  const std::vector<std::string>& accepted_suffixes)
+{
+    if (accepted_suffixes.empty()) {
+        throw std::invalid_argument(
+          "Parameter acceptedSuffixes must not be empty");
+    }
+    for (const auto& suffix : accepted_suffixes) {
+        if (suffix.empty()) {
+            throw std::invalid_argument(
+              "Parameter acceptedSuffixes must not contain empty strings");
+        }
+        if (load_from_working_directory(config_file_prefix + "." + suffix)) {
+            return true;
+        }
+    }
+    return false;
+}
+bool
+abstract_config_locator::load_from_system_property(
+  const std::string& property_key,
+  const std::vector<std::string>& accepted_suffixes)
+{
+    return load_from_system_property(property_key, false, accepted_suffixes);
+}
+void
+abstract_config_locator::load_system_property_file_resource(
+  const std::string& config_system_property)
+{
+    configuration_file = fopen(config_system_property.c_str(), "r");
+    std::cout << "Using configuration file at " << config_system_property
+              << std::endl;
+    if (configuration_file == nullptr) {
+        std::string msg =
+          "Config file at " + config_system_property + " doesn't exist.";
+        throw hazelcast::client::exception::hazelcast_(msg);
+    }
+    in->open(config_system_property);
+    if (in->fail()) {
+        throw hazelcast::client::exception::hazelcast_("Failed to open file: " +
+                                                       config_system_property);
+    }
+}
+
+bool
+abstract_config_locator::load_from_system_property(
+  const std::string& property_key,
+  bool fail_on_unaccepted_suffix,
+  const std::vector<std::string>& accepted_suffixes)
+{
+    if (accepted_suffixes.empty()) {
+        throw hazelcast::client::exception::illegal_argument(
+          "Parameter acceptedSuffixes must not be empty");
+    }
+    try {
+        std::string config_system_property =
+          "temp"; // System.getProperty(propertyKey); TODO
```

Comment:
?

Reply:
I don't know how to get property using a string